src/librustc_codegen_llvm/back/bytecode.rs

```diff
@@ -106,39 +107,39 @@ pub struct DecodedBytecode<'a> {
 }
 
 impl<'a> DecodedBytecode<'a> {
-    pub fn new(data: &'a [u8]) -> Result<DecodedBytecode<'a>, String> {
+    pub fn new(data: &'a [u8]) -> Result<DecodedBytecode<'a>, Cow<'static, str>> {
```

Comment:
I think if we're going to go this route we should instead refactor this to have an error enum with appropriate variants? In either case this is possibly worse performance-wise since it increases the size of the return value for relatively little benefit

Reply:
Actually here `Cow` is not even needed; would a `&'static str` error be ok?

Reply:
Yes, that seems fine -- if you want to open a "easy" issue, maybe even E-mentor, converting this to an error enum that'd be great. I think it doesn't help much but feels a little more rustic.